lib/src/settings.rs

```diff
@@ -45,21 +45,18 @@ impl UserSettings {
         UserSettings { config, timestamp }
     }
 
-    pub fn with_toml_strings(
-        &self,
+    pub fn incorporate_toml_strings(
+        &mut self,
         toml_strs: &[String],
-    ) -> Result<UserSettings, config::ConfigError> {
+    ) -> Result<(), config::ConfigError> {
         let mut config_builder = config::Config::builder().add_source(self.config.clone());
```

Comment:
Just fyi. I have unsent PR that adds support for loading `.jj/repo/config.toml` #616. Since repo config and `--config-toml` will have to override the base user config in that order, I'm thinking of moving `with_toml_strings()` to CLI layer. Conceptually, it will be like:

```rust
// in parse_args()
let config = override_by_file(settings.config(), ".jj/repo/config.toml");
let config = override_by_strs(config, args.config_toml);
UserSettings::from_config(config)
```

I think this will work so long as RNG is created eagerly at `UserSettings::from_config(config)`, and doesn't need to persist across UserSettings instances.

I'll rebase my change on top of this PR.

Reply:
>  I have unsent PR that adds support for loading .jj/repo/config.toml

Neat!

> I think this will work so long as RNG is created eagerly at UserSettings::from_config(config), and doesn't need to persist across UserSettings instances.

I'm not sure how well that will work with my recent changes, but I think it should be workable one way or another. Let me know, or feel free to change the relevant part of the code I created here.